src/components/news-list/news-list.js

```diff
@@ -1,51 +1,62 @@
-import React, { Component } from "react";
+import React, { useEffect, useState } from "react";
 
-import NewsService from "../../services/news-service";
+import { getIDBestNews } from "../../services/news-service";
 import NewsListItem from "../news-list-item";
 import "./news-list.css";
 
-// import Spinner from "../spinner/spinner";
-
-export default class NewsList extends Component {
-  newsService = new NewsService();
-
-  state = {
-    newsList: null,
-  };
-
-  componentDidMount() {
-    this.newsService.get20News().then((newsList) => {
-      this.setState({
-        newsList,
-      });
+import Spinner from "../spinner/spinner";
+import Pagination from "../pagination";
+
+const NewsList = () => {
+  const [newsIDBestNews, setnewsIDBestNews] = useState([]);
+  const [loading, setloading] = useState(false);
+  const [maxNumberPage, setmaxNumberPage] = useState(101);
+  const itemsPerPage = 20;
+  const [currentPage, setCurrentPage] = useState(0);
+  const onPreviousPage = () => setCurrentPage((page) => page - 1);
+  const onNextPage = () => setCurrentPage((page) => page + 1);
+  const [itemsToShow, setItemsToShow] = useState([]);
+
+  // getIDBestNews
+  useEffect(() => {
+    getIDBestNews().then((data) => {
+      console.log(data);
+      setnewsIDBestNews(data);
+      setmaxNumberPage(Math.ceil(499 / itemsPerPage) - 1);
+      setloading(true);
     });
+  }, []);
+
+  // Update items to show if page changed or newsIDBestNews updated
+  useEffect(() => {
+    const startIndex = currentPage * itemsPerPage;
+    const endIndex = startIndex + itemsPerPage;
+    setItemsToShow(newsIDBestNews.slice(startIndex, endIndex));
+  }, [newsIDBestNews, currentPage, setItemsToShow]);
+
+  // render
+  if (!loading) {
```

Comment:
`loading` is a misleading name, because the value is "true" after loading is complete, and `<Spinner />` is show when `loading` is `false`. I suggest to init with true value and reset when data is loaded:

```js
const [loading, setloading] = useState(false);
...
useEffect(...
    setloading(false);
```